### PR TITLE
Implement Retry interval

### PR DESCRIPTION
**Pull request by jebrando**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 
  - [ ] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: iothub_client/src/iothubtransport_amqp_common.c [hunk 1]
```diff
@@ -2007,6 +2007,18 @@ IOTHUB_CLIENT_RESULT IoTHubTransport_AMQP_Common_SetOption(TRANSPORT_LL_HANDLE h
                 result = IOTHUB_CLIENT_OK;
             }
         }
+        else if (strcmp(OPTION_RETRY_INTERVAL_SEC, option) == 0)
+        {
+            if (retry_control_set_option(transport_instance->connection_retry_control, RETRY_CONTROL_OPTION_INITIAL_WAIT_TIME_IN_SECS, value) != 0)
```
Comment (danewalton) [inline iothub_client/src/iothubtransport_amqp_common.c]: wait is this defined in the repo elsewhere and it just wasn't being used?

File: iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c [hunk 2]
```diff
@@ -3733,6 +3736,55 @@ TEST_FUNCTION(IoTHubTransport_MQTT_Common_Destroy_frees_the_proxy_options)
...
+
+    STRICT_EXPECTED_CALL(IoTHubClient_Auth_Get_Credential_Type(IGNORED_PTR_ARG));
+    STRICT_EXPECTED_CALL(retry_control_set_option(IGNORED_PTR_ARG, RETRY_CONTROL_OPTION_INITIAL_WAIT_TIME_IN_SECS, IGNORED_PTR_ARG)).SetReturn(__LINE__);
+
+    // act
+    int retry_interval = 10;
+    IOTHUB_CLIENT_RESULT result = IoTHubTransport_MQTT_Common_SetOption(handle, OPTION_RETRY_INTERVAL_SEC, &retry_interval);
+
+    // assert
+    ASSERT_ARE_NOT_EQUAL(IOTHUB_CLIENT_RESULT, IOTHUB_CLIENT_OK, result);
```
Comment (danewalton) [inline iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c]: >IOTHUB_CLIENT_OK [](start = 47, length = 16)

should this be a failure?

File: iothub_client/src/iothubtransport_amqp_common.c (hunk 1 above)
Comment (jebrando) [reply iothub_client/src/iothubtransport_amqp_common.c]: Yes, this was never called so we weren't able to ever set it.

---
In reply to: [323444610](https://github.com/Azure/azure-iot-sdk-c/pull/1172#discussion_r323444610) [](ancestors = 323444610)

File: iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c (hunk 2 above)
Comment (jebrando) [reply iothub_client/tests/iothubtransport_mqtt_common_ut/iothubtransport_mqtt_common_ut.c]: No I don't want to define the failure that is coming from the module, only the fact that it doesn't equal OK.

---
In reply to: [323446389](https://github.com/Azure/azure-iot-sdk-c/pull/1172#discussion_r323446389) [](ancestors = 323446389)

**Pull request merged**